src/redux/sagas/motions/expenditures/editLockedExpenditureMotion.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  type AnyVotingReputationClient,
+  ClientType,
+  ColonyRole,
+  getPermissionProofs,
+  Id,
+} from '@colony/colony-js';
+import { BigNumber } from 'ethers';
+import { call, fork, put, takeEvery } from 'redux-saga/effects';
+
+import { ADDRESS_ZERO, APP_URL } from '~constants';
+import { ActionTypes } from '~redux/actionTypes.ts';
+import {
+  createGroupTransaction,
+  createTransaction,
+  createTransactionChannels,
+  waitForTxResult,
+} from '~redux/sagas/transactions/index.ts';
+import {
+  getColonyManager,
+  initiateTransaction,
+  getMulticallDataForUpdatedPayouts,
+  uploadAnnotation,
+  getPayoutsWithSlotIds,
+} from '~redux/sagas/utils/index.ts';
+import { type Action } from '~redux/types/index.ts';
+import { type ExpenditurePayoutFieldValue } from '~types/expenditures.ts';
+import { takeFrom } from '~utils/saga/effects.ts';
+
+export type EditExpenditureMotionPayload =
+  Action<ActionTypes.MOTION_EDIT_LOCKED_EXPENDITURE>['payload'];
+
+function* editLockedExpenditureMotion({
+  payload: {
+    colonyAddress,
+    expenditure,
+    payouts,
+    networkInverseFee,
+    annotationMessage,
+    motionDomainId,
+  },
+  meta,
+  meta: { setTxHash, id },
+}: Action<ActionTypes.MOTION_EDIT_LOCKED_EXPENDITURE>) {
+  const colonyManager = yield getColonyManager();
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  const batchId = 'motion-edit-locked-expenditure';
+
+  /**
+   * @NOTE: Resolving payouts means making sure that for every slot, there's only one payout with non-zero amount.
+   * This is to meet the UI requirement that there should be one payout per row.
+   */
+  const resolvedPayouts: ExpenditurePayoutFieldValue[] = [];
```

Comment:
Could all this logic be extracted into a helper considering it's very similar to edit expenditure via permissions?